packages/babel-generator/src/source-map.ts

```diff
@@ -80,20 +106,46 @@ export default class SourceMap {
   ) {
     this._rawMappings = undefined;
 
+    let original;
+    let source;
+
+    if (line != null) {
+      if (this._inputMap) {
+        const originalMapping = originalPositionFor(this._inputMap, {
+          line,
+          column,
+        });
+
+        source =
+          originalMapping.source ||
+          filename?.replace(/\\/g, "/") ||
+          this._sourceFileName;
+
+        if (originalMapping.line != null) {
+          original = {
+            line: originalMapping.line,
+            column: originalMapping.column,
+          };
+          if (originalMapping.name) {
+            identifierName = originalMapping.name;
+          }
+        }
+      } else {
+        source = filename?.replace(/\\/g, "/") || this._sourceFileName;
+      }
+      if (!original) {
+        original = {
+          line: line,
+          column: column,
+        };
+      }
```

Comment:
There are two paths to hit this:
1. The original mapping is not contained the input map.
2. There's no input map.

I think we should only do this for 2). For 1), if the original mapping couldn't be found, then that means that this position is sourceless, and can't point to a real location. So, it's just generated, not original.

See also, [Terser's behavior](https://github.com/terser/terser/blob/4255fb3063ce1b41a5d5c9c48df443b5df0e797e/lib/sourcemap.js#L87-L111)